scripts/swagger-ui-generator/swagger-ui-generator.sh

```diff
@@ -0,0 +1,150 @@
+#!/usr/bin/env bash
+
+set -o errexit
+set -o pipefail
+
+DEFAULT_SWAGGER_UI_VERSION=v3.52.3
+SWAGGER_CONFIG_FILENAME=swagger-config.yaml
+REQUIRED_FILES=(
+  "swagger-ui-bundle.js"
+  "swagger-ui-standalone-preset.js"
+  "swagger-ui.css"
+  "favicon-16x16.png"
+  "favicon-32x32.png"
+)
+
+show_help() {
+  cat <<EOF
+Usage: $(basename "$0") <options>
+    -h, --help               Display help
+    -s, --spec-file          A file containing the API definition
+    -o, --output             The output path for the generated Swagger UI files
+    -v, --version            The Swagger-UI version to use (default: DEFAULT_SWAGGER_UI_VERSION)"
+EOF
+}
+
+main() {
+  local version="$DEFAULT_SWAGGER_UI_VERSION"
+  local output=
```

Comment:
Does this work? This looks weird, normally I would do something like:

```
function get_x {
  echo "hello"
}

function get {
  local variable=$(get_x)
}
```

It's a lot of args you are parsing but I did not know you could do this with local variables.

Reply:
I believe it's a common approach if you need to assign values for multiple variables in a single function. In this specific example `parse_command_line` is sort of a closure function (if I can say so), that is executed in scope of a parent function (`main`) and have access to its local variables.

Here is the same pattern in [helm/chart-releaser-action](https://github.com/helm/chart-releaser-action):
https://github.com/helm/chart-releaser-action/blob/6eec360873ca10c8e462c87ad50c38d1a4332638/cr.sh#L87